src/main/java/io/nats/client/impl/MessageQueue.java

```diff
@@ -71,15 +74,22 @@ class MessageQueue {
 
         this.filterLock = new ReentrantLock();
         
-        this.singleThreadedReader = singleReaderMode;
+        this.singleReaderMode = singleReaderMode;
+        this.requestCleanupInterval = requestCleanupInterval;
+    }
+
+    MessageQueue(MessageQueue source) {

```

Comment:
This allows a message queue to be created from another message queue. It is used when a writer is failing because it's blocked or a force reconnect, to take the write queue from the writer that is being discard and moving all the items (messages) in the queue to the queue in the new writer so they are not lost.